tests/core/test_case.py

```diff
@@ -357,6 +366,66 @@ def test_expect_and_assert_case_run(self):
 
         test_resource = case.all_resources[self.DEMO_RESOURCE_NAME]
 
+        self.assertTrue(isinstance(test_resource, DemoResource),
+                "State resource type should have been 'DemoResource'")
+
+        self.assertEqual(test_resource.mode,
+                         DemoResourceData.BOOT_MODE,
+                         "State resource mode attribute should "
+                         "have been 'boot'")
+
+        test_resource = DemoResourceData.objects.get(name=RESOURCE_NAME)
+
+        self.validate_resource(test_resource)
+
+    def test_expect_raises_and_assert_case_run(self):
+        """Test a TestCase that uses both expect raises and assert.
+
+        * Defines the registered resource as required resource.
+        * Runs the test under a test suite.
+        * Validates that the test failed with three tracebacks.
+        * Validates the case's data object.
+        * Validates the resource's state.
+        """
+        TempExpectRaisesCase.resources = (request('test_resource',
```

Comment:
isnt this redundant? you already requested resources in the TempExpectRaisesCase